src/libxtp/pmdecomposition.cc

```diff
@@ -0,0 +1,123 @@
+/*
+ *            Copyright 2009-2020 The VOTCA Development Team
+ *                       (http://www.votca.org)
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License")
+ *
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *              http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "votca/xtp/pmdecomposition.h"
+#include "votca/xtp/aomatrix.h"
+#include <votca/tools/eigenio_matrixmarket.h>
+
+namespace votca {
+namespace xtp {
+void PMDecomposition::computePMD(Orbitals &orbitals) {
+  Eigen::MatrixXd mo_coeff = orbitals.MOs().eigenvectors().leftCols(
+      orbitals.getNumberOfAlphaElectrons());
+  QMMolecule mol = orbitals.QMAtoms();
+  basis.Load(orbitals.getDFTbasisName());
+  aobasis.Fill(basis, mol);
+  AOOverlap overlap;
+  overlap.Fill(aobasis);
+  Eigen::MatrixXd S = overlap.Matrix();
+  double diff_D = 10000;
```

Comment:
where is this number coming from and should it be an integer?

Reply:
if you need a large value use std::numerical_limits<double>::max

Reply:
Done